### ISSUE TITLE
first_completed_at, last_completed_at, last_worked_at

### ISSUE THREAD

**Issue by jpslav** [openstax/tutor-server#460]
In order to support information about if a task was completed late, do the following:

1. Replace `TaskStep#completed_at` with `first_completed_at` and `last_completed_at`.  
2. In `TaskStep#complete` do:
  
  ```ruby
  def complete(completion_time: Time.now)
    self.first_completed_at ||= completion_time
    self.last_completed_at = completion_time
  end
  ```
3. Add `first_complete_at` and `last_completed_at` readable properties to the task step representer (`task_step_properties.rb`).
3. Add a `last_worked_at` to `Tasks::Models::Task`.  Update this to the latest completion time of a task step every time a task step is completed.  
4. Update the task representer to include `last_worked_at`.  
5. Make sure the `last_worked_at` is included in `DashboardRepresenter::TaskBase`, 
6. In `PerformanceReportRepresenter::StudentData`, add `due_at` and `last_worked_at`.
7. Add specs for all of this.

**Comment by philschatz** [comment]
At a cursory glance (code search) I don't see any references to `completed_at` or `completedAt` so maybe no FE sync needed?

**Comment by jpslav** [comment]
yep, all additive

**Comment by jpslav** [comment]
@kevinburleigh75 I might like your blessing on this.

**Comment by joemsak** [comment]
Everything in the original comment and feedback has been addressed, thanks team!

**Comment by joemsak** [comment]
Fixed by #498
